driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -208,14 +209,15 @@ public static <R extends DocumentExpression> R current() {
     }
 
     /**
-     * Returns the "current" value as a {@linkplain MapExpression map} value.
+     * Returns a reference to the "current"
+     * value as a {@linkplain MapExpression map} value.
      * The "current" value is the top-level document currently being processed
      * in the aggregation pipeline stage.
      *
-     * @return the resulting value
-     * @param <R> the type of the resulting value
+     * @return a reference to the current value as a map.
+     * @param <R> the type of the map's values.
      */
-    public static <R extends DocumentExpression> MapExpression<R> currentAsMap() {
+    public static <R extends Expression> MapExpression<R> currentAsMap() {
```

Comment:
We should have a test for this method, like we do for `current`.

Reply:
added in #1057 
https://github.com/mongodb/mongo-java-driver/pull/1057/files#diff-72f687508b977da46d04d15699a94df0a1ec21300101ddca46b5a514e12177d9R110